pkg/meta/repodb/dynamodb-wrapper/dynamo_wrapper.go

```diff
@@ -2057,3 +2071,246 @@ func (dwr *DBWrapper) SetUserMeta(ctx context.Context, userMeta repodb.UserData)
 
 	return err
 }
+
+func (dwr DBWrapper) createUserProfileTable() error {
+	_, err := dwr.Client.CreateTable(context.Background(), &dynamodb.CreateTableInput{
+		TableName: aws.String(dwr.UserProfileTablename),
+		AttributeDefinitions: []types.AttributeDefinition{
+			{
+				AttributeName: aws.String("Email"),
+				AttributeType: types.ScalarAttributeTypeS,
+			},
+		},
+		KeySchema: []types.KeySchemaElement{
+			{
+				AttributeName: aws.String("Email"),
+				KeyType:       types.KeyTypeHash,
+			},
+		},
+		BillingMode: types.BillingModePayPerRequest,
+	})
+
+	if err != nil && !strings.Contains(err.Error(), "Table already exists") {
+		return err
+	}
+
+	return dwr.waitTableToBeCreated(dwr.UserProfileTablename)
+}
+
+// func (dwr DBWrapper) deleteUserProfileTable() error {
+// 	_, err := dwr.Client.DeleteTable(context.Background(), &dynamodb.DeleteTableInput{
+// 		TableName: aws.String(dwr.UserProfileTablename),
+// 	})
+
+// 	if temp := new(types.ResourceNotFoundException); errors.As(err, &temp) {
+// 		return nil
+// 	}
+
+// 	return dwr.waitTableToBeDeleted(dwr.UserProfileTablename)
+// }
+
+// func (dwr DBWrapper) ResetUserProfileTable() error {
+// 	err := dwr.deleteUserProfileTable()
+// 	if err != nil {
+// 		return err
+// 	}
+
+// 	return dwr.createUserProfileTable()
+// }
+
+func (dwr DBWrapper) createAPIKeyTable() error {
+	_, err := dwr.Client.CreateTable(context.Background(), &dynamodb.CreateTableInput{
+		TableName: aws.String(dwr.APIKeyTablename),
+		AttributeDefinitions: []types.AttributeDefinition{
+			{
+				AttributeName: aws.String("HashedKey"),
+				AttributeType: types.ScalarAttributeTypeS,
+			},
+		},
+		KeySchema: []types.KeySchemaElement{
+			{
+				AttributeName: aws.String("HashedKey"),
+				KeyType:       types.KeyTypeHash,
+			},
+		},
+		BillingMode: types.BillingModePayPerRequest,
+	})
+
+	if err != nil && !strings.Contains(err.Error(), "Table already exists") {
+		return err
+	}
+
+	return dwr.waitTableToBeCreated(dwr.APIKeyTablename)
+}
+
+// func (dwr DBWrapper) deleteAPIKeyTable() error {
+// 	_, err := dwr.Client.DeleteTable(context.Background(), &dynamodb.DeleteTableInput{
+// 		TableName: aws.String(dwr.APIKeyTablename),
+// 	})
+
+// 	if temp := new(types.ResourceNotFoundException); errors.As(err, &temp) {
+// 		return nil
+// 	}
+
+// 	return dwr.waitTableToBeDeleted(dwr.APIKeyTablename)
+// }
+
+// func (dwr DBWrapper) ResetAPIKeyTable() error {
+// 	err := dwr.deleteAPIKeyTable()
+// 	if err != nil {
+// 		return err
+// 	}
+
+// 	return dwr.createAPIKeyTable()
+// }
+
+func (dwr DBWrapper) AddUserAPIKey(hashedKey string, email string, apiKeyDetails *repodb.APIKeyDetails) error {
+	_, err := dwr.Client.UpdateItem(context.TODO(), &dynamodb.UpdateItemInput{
+		ExpressionAttributeNames: map[string]string{
+			"#EM": "Email",
```

Comment:
Let's not hardcode 'email'. This could be a username for local auth. Can we use `identity`?